eventClass/aeroCubeEvent.py

```diff
@@ -17,23 +20,59 @@ class AeroCubeEvent(metaclass=ABCMeta):
         _INVALID_PAYLOAD_NOT_BUNDLE
     )
 
-    def __init__(self, bundle):
+    def __init__(self, bundle, signal, created_at=None, id=None):
         """
         set created_at timestamp to time.time(), e.g., the time
         since the "Epoch" (see https://en.wikipedia.org/wiki/Unix_time)
         """
-        self._created_at = time.time()
+        self._created_at = created_at if created_at is not None else time.time()
         self._payload = bundle if bundle is not None else Bundle()
+        # Check if is_valid_signal
+        self.signal = signal
+        # Hex string of deterministic uuid
+        self._uuid = id if id is not None else \
+            uuid.uuid5(uuid.NAMESPACE_OID, "{}-{}-{}".format(self.__class__.__name__, self._signal, self._created_at)).hex
 
     def __eq__(self, other):
-        return isinstance(other, self.__class__) and \
-               self._payload == other.payload and \
-               self._signal == other.signal and \
-               self._created_at == other.created_at
+        return isinstance(other, self.__class__) and self._uuid == other.uuid
 
     def __ne__(self, other):
         return not self.__eq__(other)
 
+    def __str__(self):
+        dict = {
+            'signal': str(self._signal),
+            'created_at': self._created_at,
+            'payload': str(self._payload),
+            'class': str(self.__class__.__name__),
+            'uuid': self._uuid
+        }
+        return json.dumps(dict)
+
```

Comment:
Should define a to_json function that returns the object in json representation. __str__ Should not be used for this.